includes/CreateWikiRegexConstraint.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace Miraheze\CreateWiki;
+
+use StringUtils;
+
+class CreateWikiRegexConstraint {
+
+	/**
+	 * @param array $regexes
+	 * @return bool
+	 */
+	private static function validateRegexes( $regexes ) {
+		foreach ( $regexes as $regex ) {
+			if ( !StringUtils::isValidPCRERegex( $regex ) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * @param array &$regexes
+	 */
+	private static function filterInvalidRegexes( &$regexes ) {
+		foreach ( $regexes as $regex ) {
+			if ( !StringUtils::isValidPCRERegex( $regex ) ) {
+				wfWarn( 'Contains invalid regex.' );
+
+				unset( $regexes[$regex] );
+			}
+		}
+	}
+
+	/**
+	 * @param array $lines
+	 * @return array
+	 */
+	private static function cleanLines( $lines ) {
+		return array_filter(
+			array_map( 'trim',
+				preg_replace( '/#.*$/', '', $lines )
+			)
+		);
+	}
+
+	/**
+	 * @param string $text
+	 * @return array
+	 */
+	private static function regexesFromText( $text ) {
+		$lines = explode( "\n", $text );
+		$regexes = self::cleanLines( $lines );
+
+		if ( !self::validateRegexes( $regexes ) ) {
+			wfWarn( 'Contains invalid regex.' );
+		}
+
+		return $regexes;
+	}
+
+	/**
+	 * @param array $regexes
+	 * @param string $start
+	 * @param string $end
```

Comment:
Can these descriptions be improved? I think It is not easy to understand the context of the 'start' and 'end' by just reading their names.

Reply:
Done. Thanks!